syntax/parser.go

```diff
@@ -43,6 +44,8 @@ func optionFromCode(ch rune) RegexOptions {
 		return Debug
 	case 'e', 'E':
 		return ECMAScript
+	case 'o', 'O':
```

Comment:
Do we really want patterns to be able to change the output of the `Groups()` function?  It seems like the programmer creating the `regexp` struct would be expecting certain behavior and allowing patterns to change it could be surprising (especially if the patterns are dynamic in any way).

Also, the `o` modifier has a meaning in PCRE apparently so if we decide to keep the modifier concept then we should use something besides `o`.  https://stackoverflow.com/questions/22937618/reference-what-does-this-regex-mean/22944075#22944075

Reply:
I think you are right, I removed the inline option.